### PR TITLE
Update kube-vip for InPlace upgrades.

### PR DESCRIPTION
**Pull request by rahulbabu95**
*Issue #, if available:*
Kube-vip is deployed as static pod which means during any eksd/k8s version upgrade InPlace, we need to also update the kube-vip manifest with the new image for InPlace upgrades. This change parses the udpated Kube-vip manifest from `KubeadmControlPlaneSpec.KubeadmConfigSpec` and uses a config map mounted to the upgrader pod to facilitate kube-vip update for InPlace.  The upgrader [script](https://github.com/aws/eks-anywhere-build-tooling/pull/2926) then uses the volume mount to update the kube-vip manifest onto the host. 
 
*Description of changes:*

*Testing (if applicable):*
Tested the changes locally. 

*Documentation added/planned (if applicable):*

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: controllers/controlplaneupgrade_controller.go [hunk 1]
```diff
@@ -363,3 +367,52 @@ func getCapiMachine(ctx context.Context, client client.Client, nodeUpgrade *anyw
...
+	}
+	return &corev1.ConfigMap{
+		TypeMeta: metav1.TypeMeta{
+			Kind:       "ConfigMap",
+			APIVersion: "v1",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      constants.KubeVipConfigMapName,
+			Namespace: constants.EksaSystemNamespace,
+			OwnerReferences: []metav1.OwnerReference{{
```
Comment (vivek-koppuru) [inline controllers/controlplaneupgrade_controller.go:409]: This is good but this doesn't ensure it gets cleaned up right away right? Meaning, if a new CPUpgrade needs to get created with the same name due to some subsequent upgrade, we will never get the new CM for this if it never gets deleted?

Comment (rahulbabu95) [reply controllers/controlplaneupgrade_controller.go:409]: That's a good point. I will clean it up once all the nodes are upgraded for a given cpu. This way we ensure we always clean it after a successful upgrade.

Comment (abhinavmpandey08) [reply controllers/controlplaneupgrade_controller.go:431]: I think when you have `BlockOwnerDeletion` set on the child object, it blocks the parent object deletion. @rahulbabu95 you can test it out just to be sure

File: controllers/controlplaneupgrade_controller.go [hunk 2]
```diff
@@ -363,3 +371,71 @@ func getCapiMachine(ctx context.Context, client client.Client, nodeUpgrade *anyw
 	}
 	return machine, nil
 }
+
+func cleanupKubeVipCM(ctx context.Context, log logr.Logger, client client.Client) error {
+	cm := &corev1.ConfigMap{}
+	if err := client.Get(ctx, GetNamespacedNameType(constants.KubeVipConfigMapName, constants.EksaSystemNamespace), cm); err != nil {
+		if apierrors.IsNotFound(err) {
+			log.Info("config map %s  not found, skipping deletion", "ConfigMap", constants.KubeVipConfigMapName, "Namespace", constants.EksaSystemNamespace)
```
Comment (abhinavmpandey08) [inline controllers/controlplaneupgrade_controller.go:379]: nit
```suggestion
			log.Info("config map %s not found, skipping deletion", "ConfigMap", constants.KubeVipConfigMapName, "Namespace", constants.EksaSystemNamespace)
```

File: controllers/controlplaneupgrade_controller.go [hunk 3]
```diff
@@ -363,3 +371,71 @@ func getCapiMachine(ctx context.Context, client client.Client, nodeUpgrade *anyw
...
+	if err != nil {
+		return nil, err
+	}
+	var kubeVipConfig string
+	for _, file := range kcpSpec.KubeadmConfigSpec.Files {
+		if file.Path == kubeVipStaticPodPath {
+			kubeVipConfig = file.Content
+			break
+		}
+	}
```
Comment (abhinavmpandey08) [inline controllers/controlplaneupgrade_controller.go:421]: can you add a check here to see `kubeVipConfig` is not empty after the for loop? incase the kube-vip is not found in files, just error out

File: pkg/constants/constants.go [hunk 4]
```diff
@@ -48,6 +48,9 @@ const (
 	EksaPackagesName       = "eksa-packages"
 	// UpgraderConfigMapName is the name of config map that stores the upgrader images.
 	UpgraderConfigMapName = "in-place-upgrade"
+	// KubeVipConfigMapName is the name of config map that stores the kube-vip config.
+	KubeVipConfigMapName = "kube-vip-in-place-upgrade"
+	KubeVipManifestName  = "kube-vip.yaml"
```
Comment (abhinavmpandey08) [inline pkg/constants/constants.go:53]: nit: add comment for `KubeVipManifestName`

**Review by abhinavmpandey08 — APPROVED** [review]

**Review by vivek-koppuru — APPROVED** [review]

**Pull request merged**